data/createPages/index.js

```diff
@@ -37,62 +41,86 @@ const createPages = async ({ graphql, actions: { createPage, createRedirect } })
       }
     }
   `);
-  if (result.data.allError.nodes && result.data.allError.nodes.length > 0) {
+  if (documentResult.data.allError.nodes && documentResult.data.allError.nodes.length > 0) {
     process.exit(1);
```

Comment:
We could log here, the reason we don't right now is that the error should already have been caught and logged.
Because Gatsby spawns a bunch of `async` processes during the `onCreateNode` step, though, and errors don't prevent the onCreateNode processes from continuing, the workaround is to create Error nodes and then quit when we get to the next synchronous stage in the Gatsby pipeline if any Error nodes exist.
It's an awkward solution and an alternative would be welcome, I'm not sure what we would do instead though. We could log only during createPages and never during onCreateNode, but then if there's an error that _does_ manage to crash Gatsby somehow or the Error Node is broken, we wouldn't see the output.

For context:
The original ticket here was [EDX-21](https://ably.atlassian.net/browse/EDX-21). When I originally detected missing meta descriptions, I wanted to console.warn and continue just so editors could see the error. However it was decided that it should be a requirement to always have a meta description, so the app had to avoid building while still throwing & logging an error when a page didn't have a meta description.

Reply:
thanks @MymmiJ Could we add your explanation as a comment? I think it will be quite helpful in the future.